src/network/electrum.rs

```diff
@@ -128,32 +127,187 @@ impl ElectrumConfig {
             url: electrum_url,
         }
     }
-    // Get a copy of the network (Chain) field.
-    pub fn network(&self) -> Chain {
+
+    pub fn build_bitcoin_client(&self) -> Result<ElectrumBitcoinClient, Error> {
+        ElectrumBitcoinClient::new(self.url.clone(), self.timeout, self.network)
+    }
+
+    pub fn build_liquid_client(&self) -> Result<ElectrumLiquidClient, Error> {
+        ElectrumLiquidClient::new(self.url.clone(), self.timeout, self.network)
+    }
```

Comment:
Not sure we need these functions on the config. It looks like it is possible for users to build the wrong client without any error.
Do you see a drawback for removing these methods? The specific client creation looks pretty simple even without them.

Reply:
I think the "factory-like" aspect of them provides convenience, but I agree they aren't essential. Maybe the root issue is having a single config struct for both networks? I could split the config struct into Bitcoin and Liquid counterparts

Reply:
@danielgranhao  It seem to me that the ElectrumConfig is only used to create the client. Perhaps we can get rid of the config now that we have clients?

Reply:
Done in [bd904fc](https://github.com/SatoshiPortal/boltz-rust/pull/91/commits/bd904fcef332360899fcfbf1b2e237977f80f864)

Reply:
You're right, thanks. Removed it in [73a41a3](https://github.com/SatoshiPortal/boltz-rust/pull/91/commits/73a41a3d3f2a08b6442f57f77598cc00aa639429)